moran.py

```diff
@@ -18,57 +24,102 @@ def output_players(players, outfilename="players.csv"):
         writer = csv.writer(csvfile)
         writer.writerows(rows)
 
-def build_population(players, weights):
+def build_population(i, j, weights):
+    # players = [s() for s in selected_strategies()]
+    sub_players = players[i], players[j]
     population = []
-    for player, weight in zip(players, weights):
+    for player, weight in zip(sub_players, weights):
         for _ in range(weight):
             population.append(player.clone())
     return population
 
-def run_simulations(players, N=2, turns=100, repetitions=1000, noise=0, outfilename=None):
+def write_winner(outfilename, turns, noise, names_inv, repetitions,
+                 N, i, j, seed=None):
+    """
+    Write the winner of a Moran process to file
+    """
+    if seed:
+        axl.seed(seed)  # Seed the process
+    initial_population = build_population(i, j, [1, N-1])
+
+    s1 = str(players[i].clone())
+    s2 = str(players[j].clone())
+
+    # Pull out just the interaction we need
+    outcomes = dict()
+    for pair in [(s1, s1), (s1, s2), (s2, s1), (s2, s2)]:
```

Comment:
I wasn't sure which way is faster, and when I load the data from disk I loads (s2, s1) from the (s1, s2) data.